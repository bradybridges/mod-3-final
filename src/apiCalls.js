export const getOrders = () => {
  return fetch('http://localhost:3001/api/v1/orders')
      .then(response => response.json())
}

export const addOrder = async (name, ingredients) => {
  const options = {
    method: 'POST',
    body: JSON.stringify({
      name,
      ingredients,
    }),
    headers: {
      'Content-Type': 'application/json'
    }
  }; 
  
  const result = await fetch('http://localhost:3001/api/v1/orders', options);
  return result;
}


