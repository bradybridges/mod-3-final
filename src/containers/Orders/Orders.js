import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setOrders } from '../../actions/index';
import { getOrders, deleteOrder } from '../../apiCalls';
import './Orders.css';

export class Orders extends Component {

  componentDidMount() {
    getOrders()
      .then(data => this.props.setOrders(data.orders))
      .catch(err => console.error('Error fetching:', err));
  }

  handleDelete = (e) => {
    e.preventDefault();
    const order = this.props.orders.find(order => order.name === e.target.id);
    const orderIndex = this.props.orders.findIndex(order => order.name === e.target.id);
    let updatedOrders = this.props.orders.map( order => order);
    updatedOrders.splice(orderIndex, 1);
    deleteOrder(order.id);
    this.props.setOrders(updatedOrders);
  }

  render() {
    const orderEls = this.props.orders.map(order => {
      return (
        <div className="order" key={order.name}>
          <h3>{order.name}</h3>
          <ul className="ingredient-list">
            {order.ingredients.map(ingredient => {
              return <li key={ingredient}>{ingredient}</li>
            })}
          </ul>
          <button id={order.name} onClick={this.handleDelete}>Delete</button>
        </div>
      )
    });
    return (
      <section>
        { orderEls.length ? orderEls : <p>No orders yet!</p> }
      </section>
    );
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);