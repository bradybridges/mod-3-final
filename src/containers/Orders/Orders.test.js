import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapDispatchToProps, mapStateToProps } from './Orders';
import * as api from '../../apiCalls';
import { setOrders } from '../../actions/index';

describe('Orders', () => {
  let wrapper;
  const mockOrders = [
    { name: 'Brady', id: 1, ingredients: ['lettuce'] },
    { name: 'Alan', id: 2, ingredients: ['lettuce', 'salsa', 'cheese'] },
  ];
  beforeEach(() => {
    api.getOrders = jest.fn().mockImplementation(() => {
      return Promise.resolve({ orders: mockOrders });
    });
    wrapper = shallow(<Orders orders={mockOrders} setOrders={jest.fn()} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('componentDidMount should call getOrders and setOrders', () => {
    expect(api.getOrders).toHaveBeenCalled();
    expect(wrapper.instance().props.setOrders).toHaveBeenCalledWith(mockOrders);
  });
});

describe('mapStateToProps', () => {
  it('should map orders to props', () => {
    const mockState = {
      orders: [{ name: 'Brady', id: 1, ingredients: ['lettuce'] }],
      error: 'An error happened',
    }
    const mappedState = mapStateToProps(mockState);
    expect(mappedState).toEqual({orders: [{ name: 'Brady', id: 1, ingredients: ['lettuce'] }]});
  });
});

describe('mapDispatchToProps', () => {
  const mockDispatch = jest.fn();
  const mockOrders = [{ name: 'Brady', id: 1, ingredients: ['lettuce'] }];
  const actionToDispatch = setOrders(mockOrders);
  const mappedDispatch = mapDispatchToProps(mockDispatch);
  mappedDispatch.setOrders(mockOrders);
  expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
});