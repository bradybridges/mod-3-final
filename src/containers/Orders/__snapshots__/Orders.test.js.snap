// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="Brady"
  >
    <h3>
      Brady
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="lettuce"
      >
        lettuce
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="Alan"
  >
    <h3>
      Alan
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="lettuce"
      >
        lettuce
      </li>
      <li
        key="salsa"
      >
        salsa
      </li>
      <li
        key="cheese"
      >
        cheese
      </li>
    </ul>
  </div>
</section>
`;
